### PR TITLE
Fixes genetics disks (#3657)

### PR DESCRIPTION
**Pull request by VampyrBytes**
The dna/record on the disk was a reference to the one in the buffer, not
a copy, so if you changed the one in the buffer, the one on the disk
changed.
Fixed it for cloning too

BTW, yes, I know I left in an implied src in one of the lines I changed.
This is because whoever created that file is obsessed with src, and
unless I fancied going through the file and stripping out hundreds of
them, taking out a few would actually detract from the readability of
the code

### CODE REVIEW

**Comment by Gummin** [comment]
I was going to comment something in my usual autistic manner but fuck it, 
thanks. :+1:

File: code/game/dna/dna_modifier.dm [hunk 1]
```diff
@@ -36,6 +36,18 @@
...
+	var/datum/dna2/record/newrecord = new /datum/dna2/record
+	newrecord.dna = dna.Clone()
+	newrecord.types = types
+	newrecord.name = name
+	newrecord.mind = mind
+	newrecord.ckey = ckey
+	newrecord.languages = languages
+	newrecord.implant = implant
+	return newrecord
+
```
Comment (Alex6511) [inline code/game/dna/dna_modifier.dm]: You should fix the extra white space here

**Comment by Crazylemon64** [comment]
You should probably use your copy proc for loading a buffer from the disk, too, as well as buffer injectors.

**Comment by VampyrBytes** [comment]
Good points

**Comment by TheDZD** [comment]
Future note, might want to make sure to stick a line in the PR description that uses one of the [automatic issue closing words](https://help.github.com/articles/closing-issues-via-commit-messages/#closing-an-issue-in-the-same-repository) followed by the issue number.

**Comment by VampyrBytes** [comment]
Ah right, will do

**Pull request merged**